### PR TITLE
Release guide updates based on 2.12-RC1 release process

### PR DESCRIPTION
**Pull request by tbarsballe**

### CODE REVIEW

File: doc/en/developer/source/release-guide/index.rst [hunk 1]
```diff
@@ -250,19 +260,19 @@ Post the Documentation
...
-        ProxyPass /stable/ http://ares.boundlessgeo.com/geoserver/2.9.x/doc/
-        ProxyPassReverse /stable/ http://ares.boundlessgeo.com/geoserver/2.9.x/doc/
-        ProxyPass /latest/ http://ares.boundlessgeo.com/geoserver/master/doc/
-        ProxyPassReverse /latest/ http://ares.boundessgeo.com/geoserver/master/doc/
+        ProxyPass /maintain/ https://build.geoserver.org/geoserver/2.8.x/doc/
+        ProxyPassReverse /maintain/ https://build.geoserver.org/geoserver/2.8.x/doc/
+        ProxyPass /stable/ https://build.geoserver.org/geoserver/2.9.x/doc/
+        ProxyPassReverse /stable/ https://build.geoserver.org/geoserver/2.9.x/doc/
+        ProxyPass /latest/ https://build.geoserver.org/geoserver/master/doc/
+        ProxyPassReverse /latest/ https://build.geoserver.org/geoserver/master/doc/
```
Comment (tbarsballe) [inline doc/en/developer/source/release-guide/index.rst]: This is not yet the case, due to an SSL issue between docs.geoserver.org and build.geoserver.org. That should be fixed soon, so we should hold off merging until then.

**Comment by bencaradocdavies** [comment]
Thanks @tbarsballe for these fixes and for the tidy RC release.

File: doc/en/developer/source/release-guide/index.rst [hunk 2]
```diff
@@ -121,17 +121,27 @@ When creating the first release candidate of a series, there are some extra step
...
+  * create new jobs, copying from the existing master jobs, editing the branch and the ver=master configuration.
+
+* Update the cite tests on build.geoserver.org:
+  * disable the maintenance jobs, and remove them from the geoserver view
+  * create new jobs, copying from the existing master jobs, editing the branch in the build command.
+
 * Announce on the developer mailing list that the new stable branch has been created and that the feature freeze on master is over
 
-* Switch to the new branch and update:
+* Switch to the new branch and update the documentation links, replacing ``docs.geoserver.org/latest`` with ``docs.geoserver.org/2.12.x`` (for example):
```
Comment (jodygarnett) [inline doc/en/developer/source/release-guide/index.rst]: We have to make this change for `docs.geoserver.org/maintain` as well.

**Comment by jodygarnett** [comment]
We have a small glitch in how the website is updated during this release procedure:

The geoserver.org and http://geoserver.org/download/ pages show 2.12-RC1 as the current stable release. This is incorrect as RC1 is a development release (not intended for production use). Stable (and maintenance) should always point to a production ready download.

**Pull request merged**